mqemitter-mongodb.js

```diff
@@ -40,44 +44,49 @@ function MQEmitterMongoDB (opts) {
       if (err) {
         return that.status.emit('error', err)
       }
-
       /* eslint-disable */
       var urlParsed = urlModule.parse(that._opts.url)
       var databaseName = that._opts.database || (urlParsed.pathname ? urlParsed.pathname.substr(1) : undefined)
       databaseName = databaseName.substr(databaseName.lastIndexOf('/') + 1)
 
       that._client = client
       that._db = client.db(databaseName)
+
       waitStartup()
     })
   }
 
   this._hasStream = false
   this._started = false
 
-  function waitStartup () {
+  function waitStartup() {
     that._collection = that._db.collection(opts.collection)
-    that._collection.isCapped(function (err, capped) {
-      if (that.closed) { return }
-
-      if (err) {
-        // if it errs here, the collection might not exist
-        that._db.createCollection(opts.collection, {
-          capped: true,
-          size: opts.size,
-          max: opts.max
-        }, setLast)
-      } else if (!capped) {
-        // the collection is not capped, make it so
-        that._db.command({
-          convertToCapped: opts.collection,
-          size: opts.size,
-          max: opts.max
-        }, setLast)
-      } else {
-        setLast()
-      }
-    })
+    try {
+      that._collection.isCapped(function (err, capped) {
+        if (that.closed) { return }
+
+        if (err) {
+          // if it errs here, the collection might not exist
+          that._db.createCollection(opts.collection, {
+            capped: true,
+            size: opts.size,
+            max: opts.max
+          }, setLast)
+        } else if (!capped) {
+          // the collection is not capped, make it so
+          that._db.command({
+            convertToCapped: opts.collection,
+            size: opts.size,
+            max: opts.max
+          }, setLast)
+        } else {
+          setLast()
+        }
+      })
+
+    } catch (error) {
+      that.status.emit('error', error)
+    }
```

Comment:
This is needed because there is a bug on mongo driver v4 causing this to throw instead of return an error.

Reply:
https://www.mongodb.com/community/forums/t/nodejs-driver-v4-4-1-doesnt-handle-errors-correctly/151675

Reply:
Have you opened an issue on their tracker and not just a forum question?

Reply:
I have also opened a PR: https://github.com/mongodb/node-mongodb-native/pull/3170

And there is a ticket for it: https://jira.mongodb.org/browse/NODE-4074